app/src/logging/formatters.py

```diff
@@ -21,7 +22,8 @@ def format(self, record: logging.LogRecord) -> str:
         # logging.Formatter.format adds the `message` attribute to the LogRecord
         # see https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L690-L720
         super().format(record)
-        return json.dumps(record.__dict__, separators=(",", ":"))
+
+        return json.dumps(record.__dict__, separators=(",", ":"), default=pydantic_encoder)
```

Comment:
This doesn't need to be pydantic, I just know it works.

This solves an issue where certain values can't be serialized. UUIDs and dates in particular fail to serialize properly, which pydantic handles. We can make our own copying what it does, just was fixing it in the most direct way I was aware of

Note that this wasn't an issue in older versions of this code because the `pii_mask` method implicitly made them strings.

Reply:
If we wanted to do this a bit more directly, something like:

```
def json_encoder(obj: Any) -> Any:

    match obj:
        case str() | int() | float() | bool() | None:
            return obj
        case datetime() | date():
            return obj.isoformat()
        case Decimal():
            if obj.as_tuple().exponent >= 0:
                return int(obj)
            return float(obj)
        case Enum():
            return obj.value
        case UUID():
            return str(obj)
        case _:
            return str(obj)
```
Seems to work - from testing the only things that fall into the last case are exceptions, a class in alembic, and the `exec` issue from auditing. They all get printed semi-readable, for example the exec audit log ends up as:
`"audit.args.code_object":"<code object <module> at 0x1062a3cf0, file \"/Users/michaelchouinard/.pyenv/versions/3.11.0/lib/python3.11/encodings/hex_codec.py\"`

Reply:
I went ahead and changed it, let me know what you think